pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/FilterExportConfig.java

```diff
@@ -2,28 +2,114 @@
+        this.allowMultiple = isAllowMultiple(questionDef);
+        this.questionType = this.type;
+        if (this.questionDef != null) {
+            this.options = getOptionsForQuestion(questionDef);
+            this.questionType = (String) questionDef.get(ESObjectConstants.QUESTION_TYPE);
+        }
+        List<Map<String, Object>> childQuestions = getChildQuestions();
+        if (childQuestions != null) {
+            this.childConfigs = IntStream.range(0, childQuestions.size())
+                    .mapToObj(qNum -> new FilterExportConfig(this, childQuestions.get(qNum), qNum)).collect(Collectors.toList());
+        }
+    }
+
+    /** short constructor for getting a config for a subquestion of a composite question */
+    private FilterExportConfig(FilterExportConfig parent, Map<String,Object> childQuestion, int childIndex) {
+        this.parent = parent.getParent();
+        this.type = (String) childQuestion.get(ESObjectConstants.QUESTION_TYPE);
+        String childStableId = (String) childQuestion.get(ESObjectConstants.STABLE_ID);
+        if (StringUtils.isBlank(childStableId)) {
+            // some subquestions don't have stableIds (e.g. RELEASE_MINOR_PHYSICIAN from PanCan), we need to ensure the stableId
+            // is something unique to avoid values getting overwritten
+            childStableId = (String) childQuestion.get(ESObjectConstants.QUESTION_TEXT);
+            if (StringUtils.isNotBlank(childStableId)) {
+                childStableId = childStableId.toUpperCase().replace(" ", "_");
+            } else {
+                childStableId = "SUBQUESTION" + childIndex;
+            }
+        }
+        this.column = new ParticipantColumn(childStableId, parent.column.getTableAlias());
+        String childDisplayText = (String) childQuestion.get(ESObjectConstants.QUESTION_TEXT);
+        if (StringUtils.isBlank(childDisplayText)) {
+            childDisplayText = childStableId;
+        }
+        this.column.setDisplay(childDisplayText);
+
+        this.questionDef = childQuestion;
+        this.options = getOptionsForQuestion(childQuestion);
+    }
+
+    public boolean isAllowMultiple(Map<String, Object> questionDef) {
+        if (questionDef != null && questionDef.get(ESObjectConstants.ALLOW_MULTIPLE) != null) {
+            return (boolean) questionDef.get(ESObjectConstants.ALLOW_MULTIPLE);
+        }
+        return false;
+    }
+
+    private List<Map<String, Object>> getOptionsForQuestion(Map<String, Object> questionDef) {
+        List<Map<String,Object>> options = (List<Map<String, Object>>) questionDef.get(ESObjectConstants.OPTIONS);
+        if (questionDef.containsKey(ESObjectConstants.OPTION_GROUPS)) {
+            Object groups = questionDef.get(ESObjectConstants.OPTION_GROUPS);
+            if (groups instanceof List) {
+                for (Map<String, Object> group : (List<Map<String, Object>>) groups) {
+                    if (group.containsKey(ESObjectConstants.OPTIONS) ) {
```

Comment:
I personally prefer not to use nested if statements, since it makes code hard to maintain and hard to read, in general I replace nested if statements with guard conditions.

Reply:
can you give me an example of how you'd rewrite this?  I agree that I'd like it to be cleaner, but I'm not sure I know the best way in Java to do it

Reply:
`if (!questionDef.containsKey(ESObjectConstants.OPTION_GROUPS) || !(questionDef.get(ESObjectConstants.OPTION_GROUPS) instanceof List)) return options`

I would extract boolean expression as a method for better clarification:
`if (hasOptionGroups(questionDef)) return options;`

then for loop would go as before